test/testenv.py

```diff
@@ -101,9 +101,10 @@ def get_testid(testcase):
     return testcase.get_testid()
```

Comment:
I'm not sure that the comment about the examples is warranted. `hawkmoth_root` simply has to be redirected to an examples folder and / or the directives must include the path of the files relative to `hawkmoth_root`. I.e. it _does_ work with respect to the changes we introduce here.

Anyway, nitpicking.

Reply:
It'll work if you move all the examples to the same directory, and point `hawkmoth_root` there. But if you wanted to have examples in multiple directories, it'll break...

Reply:
My point is that this code is perfect as it is with regards to that. If we have `hawkmoth_root=test/examples`, we just need to do `.. c:autodoc:: dir1/dir2/src.c`, either way it's not a limitation with regards to what you changed here. We'd need only change `conf.py` and possibly adjust `update_examples.py` unless I've missed something.

Anyway, good merge ;)